tests/kokkos-based/overwriting_matrix_matrix_product.cpp

```diff
@@ -0,0 +1,145 @@
+
+#include "gtest_fixtures.hpp"
+#include "helpers.hpp"
+
+namespace
+{
+
+template<class A_t, class B_t, class C_t>
+void gemm_gold_solution(A_t A, B_t B, C_t C)
```

Comment:
While I see the value of having "gold solution" functions, this duplicates the sequential back-end. Would you consider just calling the sequential version of matrix_product to test the Kokkos version?

Reply:
No worries then!